Consolidates troubleshooting content into the "Returning semantic field embeddings in _source" section

kosabogi opened the pull request:
This PR merges the "Troubleshooting semantic_text fields" section into the "Returning semantic field embeddings in `_source`" section to reduce duplication and improve clarity.
It also adds an annotated example response that explains key fields (`inference_id`, `model_settings` and `embeddings`) in the `_inference_fields` object.

Related issue: https://github.com/elastic/docs-content/issues/2955 
(The scope of this issue has changed, as per [this comment](https://github.com/elastic/docs-content/issues/2955#issuecomment-3372907868) and my discussion with @Mikep86 )

kosabogi committed:
Consolidates the Troubleshooting section

kosabogi committed:
Merge branch 'main' into semantic-blend-troubleshooting-section

kosabogi committed:
Merge branch 'main' into semantic-blend-troubleshooting-section

Mikep86 reviewed (commented):
Looks pretty good overall, thank you for following up on this! I left a couple comments about some minor adjustments we should consider.

kosabogi committed:
Update docs/reference/elasticsearch/mapping-reference/semantic-text.md

Co-authored-by: Mike Pellegrini <mike.pellegrini@elastic.co>

kosabogi committed:
Test applies switch tabs

kosabogi committed:
Adds version specific examples

kosabogi committed:
Syntax fix

kosabogi committed:
Fixes syntax

kosabogi committed:
Merge branch 'main' into semantic-blend-troubleshooting-section

jimczi commented on docs/reference/elasticsearch/mapping-reference/semantic-text.md:
```diff
@@ -534,12 +535,184 @@ POST test-index/_search
+  },
+  "fields": [
+    "_inference_fields"
+  ]
```
Please use the version with `exclude_vectors` to false, this is the preferred way to retrieve the field.

jimczi replied:
Can we start with the preferred and current way of doing things? We can document the old way but it should be a small warning section without needing to repeat the entire response (it's the same).

kosabogi committed:
Adds separate tabs for each versions

kosabogi committed:
Adds only two version tahs

kosabogi committed:
Deletes tabs, uses section level applies-to instead

kosabogi commented on docs/reference/elasticsearch/mapping-reference/semantic-text.md:
```diff
@@ -526,12 +527,112 @@ POST test-index/_search
+:::{applies-item} stack: ga 9.1
+Other content for 9.1 version
+:::
+::::
```
I changed the formatting a bit, since our applies-switch functionality doesn’t support version ranges yet.
I created two separate headings - one for 9.0–9.1 and another for 9.2 - each containing the appropriate version of the recommended search query.
What do you think of this approach?

kosabogi committed:
Merge branch 'main' into semantic-blend-troubleshooting-section

kosabogi committed:
Removes response block, adds intros and warning

Mikep86 commented on docs/reference/elasticsearch/mapping-reference/semantic-text.md:
````diff
@@ -534,12 +535,128 @@ POST test-index/_search
+
+#### Retrieve embeddings from version 9.2 [retrieve-embeddings-9.2]
+```{applies_to}
+stack: ga 9.2
````
Can we also mention serverless in this `applies_to` section

Mikep86 reviewed (approved):
The content looks good, thank you for the iterations! I left one comment about mentioning serverless in the `applies_to` tag.

Tangential to this PR, I think we can make a few changes to [Example: Reindex while preserving embeddings](https://docs-v3-preview.elastic.dev/elastic/elasticsearch/pull/137233/reference/elasticsearch/mapping-reference/semantic-text#example-reindex-while-preserving-embeddings) to make it clearer:

- Use an `applies_to` tag to indicate that this example only applies to 9.2+ and serverless
- Remove the section that uses the `fields` param. The `fields` param can't be used during reindexing, so mentioning it as part of reindexing is confusing.

That change could be it's own PR though. WDYT?

kosabogi committed:
Reorganizes content, adds applies_to where necessary

kosabogi commented:
> The content looks good, thank you for the iterations! I left one comment about mentioning serverless in the `applies_to` tag.
> 
> Tangential to this PR, I think we can make a few changes to [Example: Reindex while preserving embeddings](https://docs-v3-preview.elastic.dev/elastic/elasticsearch/pull/137233/reference/elasticsearch/mapping-reference/semantic-text#example-reindex-while-preserving-embeddings) to make it clearer:
> 
> * Use an `applies_to` tag to indicate that this example only applies to 9.2+ and serverless
> * Remove the section that uses the `fields` param. The `fields` param can't be used during reindexing, so mentioning it as part of reindexing is confusing.
> 
> That change could be it's own PR though. WDYT?

Thank you, really good catches!
I applied your suggestions in this PR, as I felt it would be confusing without them.

In my latest commit, I made the following changes:

- Added an applies-to tag to the [Example: Reindex while preserving embeddings](https://docs-v3-preview.elastic.dev/elastic/elasticsearch/pull/137233/reference/elasticsearch/mapping-reference/semantic-text#example-reindex-while-preserving-embeddings) section.
- Removed the note that mentioned the fields parameter, since we now show the method for retrieving embeddings for earlier versions (pre-9.2) in a dedicated section.
- Renamed the _Returning semantic field embeddings for 9.0–9.1_ section to _Returning semantic field embeddings using fields_ to stay consistent with the _Returning semantic field embeddings in source_ title.
- Moved the _Returning semantic field embeddings using fields_ section one level higher. I did this because it makes more sense for it to be a separate section rather than part of _Returning semantic field embeddings in source_, as it describes a different method. 
- At the beginning of the  _Returning semantic field embeddings using fields_ section, I added a warning to clarify that this applies only to Elasticsearch versions earlier than 9.2, and that users on 9.2+ should refer to the other section.
- At the beginning of the _Returning semantic field embeddings in source_ section, I added a note linking to the section that explains the method for earlier versions.

What do you think about it now?

kosabogi committed:
Merge branch 'main' into semantic-blend-troubleshooting-section

Mikep86 commented on docs/reference/elasticsearch/mapping-reference/semantic-text.md:
```diff
@@ -544,9 +544,14 @@ stack: ga 9.2
 
+:::{important}
+The recommended method for retrieving embeddings differs between {{es}} versions 9.2 and earlier.
+For instructions on retrieving embeddings in versions earlier than 9.2, refer to [Retrieve semantic field embeddings using `fields`](#retrieve-embeddings-9.0-9.1).
```
The heading for this section is "Returning semantic field embeddings using `fields`". Can we keep that aligned with the text here?

```suggestion
For instructions on retrieving embeddings in versions earlier than 9.2, refer to [Returning semantic field embeddings using `fields`](#retrieve-embeddings-9.0-9.1).
```

Mikep86 reviewed (commented):
Structure looks good to me, I left a few comments about minor edits

kosabogi committed:
Update docs/reference/elasticsearch/mapping-reference/semantic-text.md

Co-authored-by: Mike Pellegrini <mike.pellegrini@elastic.co>

kosabogi committed:
Update docs/reference/elasticsearch/mapping-reference/semantic-text.md

Co-authored-by: Mike Pellegrini <mike.pellegrini@elastic.co>

kosabogi committed:
Adds additional information, changes admonition types

Mikep86 commented on docs/reference/elasticsearch/mapping-reference/semantic-text.md:
```diff
@@ -602,16 +612,109 @@ the documents will **fail the reindex task**.
+2. Lists details about the model used to generate embeddings, such as the service name and task type.
+3. The embeddings generated for this chunk.
+
+## Returning semantic field embeddings using `fields` [retrieve-embeddings-9.0-9.1]
```
Got it, thank you for explaining!

Mikep86 reviewed (approved):
LGTM, thank you for the iterations!

jimczi reviewed (approved):
Sorry I didn't meant to block the merging with my old request for changes. :LGTM

szabosteve commented on docs/reference/elasticsearch/mapping-reference/semantic-text.md:
```diff
@@ -602,16 +612,110 @@ the documents will **fail the reindex task**.
+3. The embeddings generated for this chunk.
+
+## Returning semantic field embeddings using `fields` [return-embeddings-fields]
+
```
Is there a reason not to add an applies_to tag that lists 9.0 and 9.1 and marks it unavailable from 9.2?

kosabogi replied:
Yes - I didn’t want to mark this as unavailable from 9.2, since it’s not technically unavailable, it’s just not recommended to use.
I thought specifically listing 9.0 and 9.1 might be a bit misleading, as it could imply that this parameter is only available starting from 9.0. But I’m open to adding the 9.0 and 9.1 tags here, I was thinking about that myself too. Do you think it would be better?

szabosteve reviewed (commented):
I left a few comments.

kosabogi committed:
Update docs/reference/elasticsearch/mapping-reference/semantic-text.md

Co-authored-by: István Zoltán Szabó <istvan.szabo@elastic.co>

kosabogi committed:
Shorten tab titles

kosabogi committed:
Merge branch 'main' into semantic-blend-troubleshooting-section

leemthompo reviewed (approved):
If Jim and Mike are happy, I'm happy 😄 and we'll revisit this page to break it up into logical subpages and cleanup versioning conditionals as much as possible in future work

Thx!
